lib/perf_hooks.js

```diff
@@ -467,6 +469,10 @@ function doNotify() {
 // Set up the callback used to receive PerformanceObserver notifications
 function observersCallback(entry) {
   const type = mapTypes(entry.entryType);
+
+  if (type === NODE_PERFORMANCE_ENTRY_TYPE_HTTP2)
+    collectStats(entry);
```

Comment:
Can you move this into the http2 namespace? I'm kind of -1 on having perf-hooks require something within http2 itself.

Reply:
Not following. The `collectStats` method *is* in the `internal/http2/util.js` file. It is used here because we are instrumenting the http2 performance entry object with the http2 specific details, which must be done within this function.

Reply:
The rest of the code in perf_hooks is isolated from the rest of core. We are not loading fs, http, net etc to instrument them. Why do we need to load a part of http2?

Reply:
ah, ok, so move the `collectStats` method in to perf_hooks to keep from having to require anything from http2... yeah, I can do that.

Reply:
why do we need this for http2 while we don't for http or net?

Reply:
Neither `http` or `net` *currently* output any performance metrics via the perf_hooks API. That's something that's on my lower priority todo list. This is there for http2 because we're using an aliased buffer to move those data values out to js instead of setting properties on the object from within the C++ code -- it's far more efficient to do it this way. Eventually, once other bits of code are instrumented with perf_hooks output, similar mechanisms will be added. For instance, I'm planning on doing something similar with dns lookups fairly soon.